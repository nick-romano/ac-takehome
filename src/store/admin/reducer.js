

export const reducer = (state, action) => {
    switch(action.type) {
        case 'setUser':
            return {
                ...state,
                user: action.user
            }
        default:
            throw new Error("unknown action dispatched to admin reducer");
    }
}
