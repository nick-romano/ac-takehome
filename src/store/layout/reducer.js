
export const reducer = (state, action) => {
    switch(action.type) {
        case "setAdminMenuOpen":
            return {
                ...state,
                adminMenuOpen: action.menuOpen
            }
        case "setSideMenuSelection": {
            return {
                ...state,
                sideMenuSelection: action.id
            }
        }
        case "setShelvesMenuExpanded":
            return {
                ...state,
                shelvesMenuExpanded: action.shelvesMenuExpanded
            }
        case "setAddShelfModalOpen":
            return {
                ...state,
                addShelfModalOpen: action.addShelfModalOpen
            }
        case "setNextPage":
            return {
                ...state,
                nextPage: action.nextPage
            }
        case "setChangeUserModalOpen":
            return {
                ...state,
                changeUserModalOpen: action.changeUserModalOpen
            }
        default:
            throw new Error("Unknown action type provided to admin reducer");
    }
}

