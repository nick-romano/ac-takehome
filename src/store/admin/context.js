import { createContext, useReducer } from "react";
import { reducer } from "./reducer";


export const initialState = {
    user: "blacklight"
};

const AdminContext = createContext(initialState);

const AdminProvider = ({ children }) => {

    const [state, dispatch] = useReducer(reducer, initialState);
    
    return (
        <AdminContext.Provider
            value={{
                ...state,
                dispatch
            }}
        >
            {children}
        </AdminContext.Provider>
    )
};

export { AdminProvider, AdminContext };





