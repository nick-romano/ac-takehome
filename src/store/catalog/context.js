import { createContext, useReducer } from "react";
import { reducer } from "./reducer";

export const initialState = {  
};

const CatalogContext = createContext(initialState); 

const CatalogProvider = ({ children }) => {

    const [state, dispatch] = useReducer(reducer, initialState);

    return (
        <CatalogContext.Provider
            value={{
                shelves: state,
                dispatch
            }}
        >
            {children}
        </CatalogContext.Provider>
    )

};

export { CatalogContext, CatalogProvider };