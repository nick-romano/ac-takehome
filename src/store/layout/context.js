import { createContext, useReducer } from "react";
import { reducer } from "./reducer";


export const initialState = {
    adminMenuOpen: false,
    shelvesMenuExpanded: true,
    sideMenuSelection: undefined,
    addShelfModalOpen: false,
    perPage: 25,
    nextPage: undefined,
    changeUserModalOpen: false
};

const LayoutContext = createContext(initialState);

const LayoutProvider = ({ children }) => {

    const [state, dispatch] = useReducer(reducer, initialState);
    
    return (
        <LayoutContext.Provider
            value={{
                ...state,
                dispatch
            }}
        >
            {children}
        </LayoutContext.Provider>
    )
};

export { LayoutProvider, LayoutContext };





