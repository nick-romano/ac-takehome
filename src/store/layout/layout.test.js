import { render } from '@testing-library/react';
import { reducer } from './reducer';
import { initialState } from './context';
import { LayoutProvider } from './context';

describe('testing layout store and reducer', () => {
    it('should update the state variable adminMenuOpen', () => {
        expect(initialState.adminMenuOpen).toBe(false);
        const state = reducer(initialState, {
            type: "setAdminMenuOpen",
            menuOpen: true
        });
        
        expect(state.adminMenuOpen).toBe(true);
    });

    it('should update shelvesMenuExpanded', () => {
        const state = reducer(initialState, {
            type: "setShelvesMenuExpanded",
            menuOpen: false
        });
        
        expect(state.adminMenuOpen).toBe(false);
    })

    it('should update AddShelfModalOpen', () => {
        const state = reducer(initialState, {
            type: "setAddShelfModalOpen",
            addShelfModalOpen: true
        });

        expect(state.addShelfModalOpen).toBe(true);
    })

    it('should update nextPage', () => {
        const state = reducer(initialState, {
            type: "setNextPage",
            nextPage: 'https://test/'
        });

        expect(state.nextPage).toBe('https://test/');
    })

    it('should render layout provider', () => {
        const { asFragment, getByText } = render(<LayoutProvider><div>TEST_CHILD</div></LayoutProvider>);
        expect(getByText("TEST_CHILD")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});

