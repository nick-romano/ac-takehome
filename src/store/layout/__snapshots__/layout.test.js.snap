// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing layout store and reducer should render layout provider 1`] = `
<DocumentFragment>
  <div>
    TEST_CHILD
  </div>
</DocumentFragment>
`;
